pkg/cache/server/handler.go

```diff
@@ -141,3 +142,11 @@ func WithServiceScope(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, req)
 	})
 }
+
+// WithSyntheticDelay injects a synthetic delay to calls, to exacerbate timing issues and expose inconsistent client behavior.
+func WithSyntheticDelay(handler http.Handler, delay time.Duration) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		time.Sleep(delay)
```

Comment:
Would have expected that this will slow down the replication controllers a lot because with this the throughput it heavily reduced. But it looks like it isn't (tests are green)? Do we have proof that this change is actually effective?

Reply:
@stevekuznetsov have you double checked e.g. with some logging?

Reply:
What is an outgoing request from the cache server?

Reply:
> fs.DurationVar(&o.SyntheticDelay, "synthetic-delay", 0, "The duration of time the cache server will inject a delay for to all inbound and outbound requests. Useful for testing.")

You tell me :). Fix the comment, add logging for each request and let's merge it.

Reply:
I don't think we need to spam logs per request here. Should be clear that this is working from p0 metrics.